synapse/federation/federation_server.py

```diff
@@ -148,6 +148,30 @@ def _handle_incoming_transaction(self, origin, transaction, request_time):
 
         logger.debug("[%s] Transaction is new", transaction.transaction_id)
 
+        # Reject if PDU count > 50 and EDU count > 100
+        if (len(transaction.pdus) > 50
+                or (hasattr(transaction, "edus") and len(transaction.edus) > 100)):
+            response = {
+                "pdus": {}
+            }
+
+            for pdu_key in transaction["pdus"].keys():
```

Comment:
I'm really not sure I'd bother with this. The spec doesn't say that this will be present for a non-200 response, and synapse will ignore it.

(in any case `Transaction` doesn't have a __getattr__, so `transaction["pdus"]` will fail, and the pdus are a list not a dict.)